pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseTaskExecutor.java

```diff
@@ -56,4 +57,12 @@ protected Schema getSchema(String tableName) {
     Preconditions.checkState(schema != null, "Failed to find schema for table: %s", tableName);
     return schema;
   }
+
+  protected long getSegmentCrc(String tableNameWithType, String segmentName) {
+    OfflineSegmentZKMetadata segmentZKMetadata = ZKMetadataProvider
```

Comment:
I am assuming BaseSingleSegmentConversionExecutor are only used by Offline type table. Do we need add support for realtime case? @mcvsubbu @jackjlli

Reply:
I do see some task executor like `RealtimeToOfflineSegmentsTaskExecutor` that works on realtime segments, although it extends `BaseMultipleSegmentsConversionExecutor`. I think it's a good practice to eliminate all these assumptions to make the code clean.

Reply:
done